docs/src/tutorials/diffusion_implicit_heat_equation.md

```diff
@@ -0,0 +1,186 @@
+# Solving the heat equation with diffusion-implicit time-stepping.
+
+In this tutorial, we'll be solving the heat equation:
+
+```math
+∂_t T = α ∇²(T) + β \sin(γ z)
+```
+
+with boundary conditions: ``∇T(z=a) = ∇T_{bottom}, T(z=b) = T_{top}``. We'll solve these equations on cell faces, rather than cell centers. The same exercise could easily be done on cell centers.
+
+First, we'll use / import some packages:
+
+```julia
+import Plots
+using LinearAlgebra
+using DiffEqBase
+using OrdinaryDiffEq: SplitODEProblem, solve, IMEXEuler
+import SciMLBase
+```
+
+Next, we'll define some global problem parameters:
+```julia
+a,b, n = 0, 1, 10               # zmin, zmax, number of cells
+n̂_min, n̂_max = -1, 1            # Outward facing unit vectors
+α = 100;                        # thermal diffusivity, larger means more stiff
+β, γ = 10000, π;                # source term coefficients
+Δt = 1000;                      # timestep size
+N_t = 10;                       # number of timesteps to take
+FT = Float64;                   # float type
+Δz = FT(b-a)/FT(n)
+Δz² = Δz^2;
+∇²_op = [1/Δz², -2/Δz², 1/Δz²]; # interior Laplacian operator
+∇T_bottom = 10;                 # Temperature gradient at the top
+T_top = 1;                      # Temperature at the bottom
+S(z) = β*sin(γ*z)               # source term, (sin for easy integration)
+zf = range(a, b, length=n+1);   # coordinates on cell faces
```

Comment:
Is this actually needed for the tutorial? It seems like this is doing a lot of unnecessary things. Seems more like a https://github.com/SciML/SciMLTutorials.jl tutorial instead of something streamlined for the docs.

Reply:
Ah, yeah, I wasn’t exactly sure where to put this. I’m fine with moving it (the whole tutorial, right?) there

Reply:
Yeah, let's go with the move. I am putting up a bounty soon anyways to try and find a way to incorporate those tutorials and the benchmarks into the documentation. It's mostly split so that it's easy to regenerate the results using the AMDCI computers, allowing you to go ham and make it a big equation and really demonstrate sparsity handling etc. I hope that it will soon then feed back nicely into the main docs.